static/app/views/routeContext.tsx

```diff
@@ -0,0 +1,7 @@
+import {createContext} from 'react';
+import {RouteContextInterface} from 'react-router';
+
+// TODO(nisanthan): Better types. Context will be the `props` arg from the RouterProps render method. This is typed as `any` by react-router
+export const RouteContext = createContext<RouteContextInterface>(
+  {} as RouteContextInterface
```

Comment:
Why do we init with {}? This will hide usage where we forget to wrap a provider.

I think we should init to null here and then inside useRoutes check for null
```
export const RouteContext = createContext<RouteContextInterface | null>(null)
```

function useRoutes() {
  const context = useContext(RouteContext);
  if(context === null){
     throw new Error("useRoutes called outside of routes provider")
  }

  return route.routes;
}